.github/workflows/src/sdk-breaking-change-labels.js

```diff
@@ -0,0 +1,186 @@
+// @ts-check
+import { sdkLabels } from "../../src/sdk-types.js";
+import { LabelAction } from "../../src/label.js";
+
+const wfName = 'sdk-breaking-change-labels';
+
+/**
+ * @param {import('github-script').AsyncFunctionArguments} AsyncFunctionArguments
+ * @returns {Promise<{labelName: string, labelAction: LabelAction, issueNumber: number, state: string}>}
+ */
+export async function getLabelAndAction({ github, context, core }) {
+  let issue_number = NaN;
+  let check_run_id = process.env.CHECK_RUN_ID || context.payload.check_run?.id;  
+  let labelAction;
+  let labelName = "";
+
+  // Determine check run ID to use (from manual input or hardcoded default)
+  if (!check_run_id) {
+    throw new Error('No check run ID provided');
+  }
+  core.info(`Fetching check run with ID: ${check_run_id}`);
+  
+  // First fetch the check run with specific ID
+  const checkRunData = await github.rest.checks.get({
+    owner: context.repo.owner,
+    repo: context.repo.repo,
+    check_run_id: parseInt(check_run_id, 10)
+  });
+  
+  // Use the check run data we retrieved instead of from the payload
+  const checkRun = checkRunData.data;
+  if (!checkRun) {
+    throw new Error('No check run data found in the API response');
+  }
+  core.info(`Check run name: ${checkRun.name}`);
+  if (!checkRun.name.includes("(SDK Generation Job)")) {
+    core.info(`Ignoring check run as it's not an SDK generation job`);
+    core.setOutput('result', 'neutral');
+    return { labelName, labelAction: LabelAction.None, issueNumber: issue_number, state: 'neutral' };
+  }
+
+  // Extract build ID from details_url if available
+  if (checkRun.details_url) {
+    core.info(`Details URL:, ${checkRun.details_url}`);
+    
+    // Extract buildId using regex
+    const buildIdMatch = checkRun.details_url.match(/buildId=(\d+)/);
+    if (buildIdMatch && buildIdMatch[1]) {
+      const buildId = buildIdMatch[1];
+      
+      // Extract organization and project from URL
+      const urlParts = new URL(checkRun.details_url);
+      const pathSegments = urlParts.pathname.split('/').filter(Boolean);
+      
+      if (pathSegments.length >= 2) {
+        const organization = pathSegments[0]; // e.g., azure-sdk
+        const project = pathSegments[1]; // e.g., fadad8ea-1e34-4d7a-9016-fd29b14e8b94
+
+        // Construct artifacts URL
+        const artifactsUrl = `https://dev.azure.com/${organization}/${project}/_build/results?buildId=${buildId}&view=artifacts`;
+        core.info(`Artifacts URL: ${artifactsUrl}`);
+        
+        // Call Azure DevOps REST API to list artifacts
+        try {          
+          // API URL to list artifacts
+          const apiUrl = `https://dev.azure.com/${organization}/${project}/_apis/build/builds/${buildId}/artifacts?api-version=7.0`;              
+          core.info('Calling Azure DevOps API to list artifacts...');
+          
+          // Use Node.js fetch to call the API
+          const response = await fetch(apiUrl, {
```

Comment:
Does this require authentication?  Have you tested it for private pipelines (repo specs-pr) as well as public?